src/Propel/Runtime/ActiveQuery/Criteria.php

```diff
@@ -1247,6 +1255,75 @@ public function hasSelectModifier($modifier)
         return in_array($modifier, $this->selectModifiers);
     }
 
+    /**
+     * @return \Propel\Runtime\ActiveQuery\Lock|null Get read lock value.
+     */
+    public function getLock(): ?Lock
+    {
+        return $this->lock;
+    }
+
+    /**
+     * Apply a shared read lock to be used to retrieve rows.
+     *
+     * @param string[] $tableNames
+     * @param bool $noWait
+     *
+     * @return $this Modified Criteria object (for fluent API)
+     */
+    public function forShare(array $tableNames = [], bool $noWait = false)
+    {
+        $this->withLock(Lock::SHARED, $tableNames, $noWait);
+
+        return $this;
+    }
+
+    /**
+     * Apply an exclusive read lock to be used to retrieve rows.
+     *
+     * @param string[] $tableNames
+     * @param bool $noWait
+     *
+     * @return $this Modified Criteria object (for fluent API)
+     */
+    public function forUpdate(array $tableNames = [], bool $noWait = false)
```

Comment:
As far as I know, the method name `forUpdate` is a very standard name, and some other projects already implemented this as it was necessary, this can cause BC breaking:
e.g. 
https://github.com/spryker/propel-orm/blob/master/src/Spryker/Zed/PropelOrm/Business/Builder/QueryBuilder.php#L447

 so maybe a more unique name can reduce the chance of the breaking or do you have a better idea. this is the same for `forShare`

Reply:
Hi @ehsanmx. Yes, these names are common-looking because I wanted to make the naming similar to how Propel is built in general. Originally the API was `withLock(Lock::SHARED)` but that just didn't feel right IMHO, not with other methods named so simple and similar-looking to what SQL offers in standard - that's the "why" part of this PR.

I know that such change could introduce a BC break in other projects but should we look at who and how implemented the library? Choosing other names for these methods like `selectForUpdate` may subject to the same problem - somebody somewhere may have implemented such method already - it's rather a matter of probability of such situation in other projects. Personally I don't think that it's a BC break because the API isn't changed or removed but instead a feature is added so it's a minor change (according to [Semantic Versioning](https://semver.org/spec/v2.0.0.html)). I don't know if Propel follows semantic versioning (doubt it really) although the idea of following it would be nice.

But, not to be so negative about it 😉, I would propose also `lockForShare`/`lockForUpdate` or `withSharedLock`/`withExclusiveLock`, however these don't feel as nice and close to native SQL parts as the originally proposed names. Maybe that's the same reason why currently proposed naming was also implemented in [QueryBuilder.php#L447](https://github.com/spryker/propel-orm/blob/master/src/Spryker/Zed/PropelOrm/Business/Builder/QueryBuilder.php#L447)?

Reply:
Hey @prgTW,
Thanks for your reply, I think having more keywords in the method name like `lockForShare/lockForUpdate` reduce the chance of breaking and I'm ok with that

Reply:
Naming changed as requested - `lockForShare`,`lockForUpdate`,`withoutLock` :)